pkg/discovery/discovery.go

```diff
@@ -226,3 +236,87 @@ func (d *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(mastersArr, ",")), nil
 }
+
+func (d *tidbDiscovery) VerifyPDEndpoint(advertisePeerURL string) (string, error) {
```

Comment:
```suggestion
func (d *tidbDiscovery) VerifyPDEndpoint(pdURL string) (string, error) {
```

Reply:
All the following code refers to `pdURL` from the different components, there is no `advertise` nor `peer` concept in the URL.